src/_pytest/assertion/util.py

```diff
@@ -439,7 +442,8 @@ def _compare_eq_cls(
         explanation += ["Differing attributes:"]
         for field in diff:
             explanation += [
-                ("%s: %r != %r") % (field, getattr(left, field), getattr(right, field))
+                ("%s: %r != %r") % (field, getattr(left, field), getattr(right, field)),
+                *_compare_eq_any(getattr(left, field), getattr(right, field), verbose),
```

Comment:
Instead of `%r` this could also use `pprint.saferepr`: https://github.com/python/cpython/blob/6003db7db5fec545c01923c198a5fdfca5a91538/Lib/pprint.py#L60-L62

Reply:
I didn't change those lines, so I would rather not change other things than the recursion for now.